### PR TITLE
Remove the legacy tactic compatibility layer from ssrelim

### PR DESCRIPTION
**Pull request by ppedrot**
Yet another round of cleanup! We'll end up somewhere eventually.

### CODE REVIEW

File: plugins/ssr/ssrelim.ml [hunk 1]
```diff
@@ -99,377 +101,401 @@ let subgoals_tys sigma (relctx, concl) =
...
-        str"occurs in the type of another non-instantiated pattern variable");
-    end
+  let sigma, elim_pred, gen_eq_tac, clr =
+    generate_pred env sigma0 ~concl patterns predty eqid is_rec deps elim_args n_elim_args c_is_head_p clr sigma
   in
+  let sigma, pty = Typing.type_of env sigma elim_pred in
+  debug_ssr (fun () -> Pp.(str"elim_pred=" ++ pr_econstr_env env sigma elim_pred));
+  debug_ssr (fun () -> Pp.(str"elim_pred_ty=" ++ pr_econstr_env env sigma pty));
+  let sigma = unify_HO env sigma pred elim_pred in
+  let elim = fire_subst sigma elim in
```
Comment (gares) [inline plugins/ssr/ssrelim.ml:473]: do we really need this in the new API?

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
I'd be happier if `fire_subst` was the identity. Did you try?

**Comment by ppedrot** [comment]
I didn't try but I am a bit suspicious that everything will go well given that some functions in ssrcommon are using the evar-sensitive API. This should be ported to EConstr at some point but as already explained in another PR, I am progressing cautiously towards piecewise porting / cleanup. There are a lot of weird things going on in ssr, notably regarding evar surgery and the simultaneous consideration of several evarmaps. We should strive to get a stricter API enforcing more things statically.

**Comment by gares** [comment]
@coqbot  merge now

**Comment by gares** [comment]
@coqbot merge now

**Pull request merged**